File: airbyte-cdk/java/airbyte-cdk/typing-deduping/src/main/java/io/airbyte/integrations/base/destination/typing_deduping/Struct.java
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.integrations.base.destination.typing_deduping;

import java.util.LinkedHashMap;

/**
 * @param properties Use LinkedHashMap to preserve insertion order.
 */
public record Struct(LinkedHashMap<String, AirbyteType> properties) implements AirbyteType {

}

File: airbyte-cdk/java/airbyte-cdk/typing-deduping/src/main/kotlin/io/airbyte/integrations/base/destination/typing_deduping/Struct.kt
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */
package io.airbyte.integrations.base.destination.typing_deduping

/** @param properties Use LinkedHashMap to preserve insertion order. */
data class Struct(val properties: LinkedHashMap<String, AirbyteType>) : AirbyteType {
    override val typeName: String = TYPE

    companion object {
        const val TYPE: String = "STRUCT"
    }
}
